doc/reference/configuration/configuration_reference.rst

```diff
@@ -221,6 +488,718 @@ This section describes options related to :ref:`storing configuration in etcd <c
 
 
 
-..  TODO
-    https://github.com/tarantool/doc/issues/3664
+..  _configuration_reference_database:
+
+database
+--------
+
+The ``database`` section defines database-specific configuration parameters, such as an instance's read-write mode, transaction isolation level, and so on.
+
+.. NOTE::
+
+    ``database`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+-   :ref:`database.hot_standby <configuration_reference_database_hot_standby>`
+-   :ref:`database.instance_uuid <configuration_reference_database_instance_uuid>`
+-   :ref:`database.mode <configuration_reference_database_mode>`
+-   :ref:`database.replicaset_uuid <configuration_reference_database_replicaset_uuid>`
+-   :ref:`database.txn_isolation <configuration_reference_database_txn_isolation>`
+-   :ref:`database.txn_timeout <configuration_reference_database_txn_timeout>`
+-   :ref:`database.use_mvcc_engine <configuration_reference_database_use_mvcc_engine>`
+
+.. _configuration_reference_database_hot_standby:
+
+.. confval:: database.hot_standby
+
+    | Type: boolean
+    | Default: false
+    | Environment variable: TT_DATABASE_HOT_STANDBY
+
+
+.. _configuration_reference_database_instance_uuid:
+
+.. confval:: database.instance_uuid
+
+    An :ref:`instance UUID <replication_uuid>`.
+
+    By default, instance UUIDs are generated automatically.
+    ``database.instance_uuid`` can be used to specify an instance identifier manually.
+
+    UUIDs should follow these rules:
+
+    *   The values must be true unique identifiers, not shared by other instances
+        or replica sets within the common infrastructure.
+
+    *   The values must be used consistently, not changed after the initial setup.
+        The initial values are stored in :ref:`snapshot files <index-box_persistence>`
+        and are checked whenever the system is restarted.
+
+    *   The values must comply with `RFC 4122 <https://tools.ietf.org/html/rfc4122>`_.
+        The `nil UUID <https://tools.ietf.org/html/rfc4122#section-4.1.7>`_ is not allowed.
+
+    See also: :ref:`database.replicaset_uuid <configuration_reference_database_replicaset_uuid>`
+
+    | Type: string
+    | Default: box.NULL
+    | Environment variable: TT_DATABASE_INSTANCE_UUID
+
+
+.. _configuration_reference_database_mode:
+
+.. confval:: database.mode
+
+    An instance's operating mode.
+    The following modes are available:
+
+    -   ``rw``: an instance is in read-write mode.
+    -   ``ro``: an instance is in read-only mode.
+
+    If this option is not specified explicitly, its default value depends on the number of instances in a replica set. For a single instance, the ``rw`` mode is used, while for multiple instances, the ``ro`` mode is used.
```

Comment:
It is (almost) correct for `failover: off` mode. However, the logic is different for different failover modes. Look [here](https://github.com/tarantool/tarantool/blob/2c422a0b98675cab480ebb44256dfeb72bd2f820/src/box/lua/config/applier/box_cfg.lua#L347-L487). It is a bit complicated and, maybe, requires some separate effort to explain to users.

Why almost?

There is the special logic on startup, which (in some cases, when it is possible) starts an instance that is configured as RW in the RO mode. After that it re-reads the configuration and if the instance still configured as RW, goes to RW. The idea is to prevent two masters in the off/manual failover modes.

Say, an administrator wants to recover a master, which want off beforehand. The administrator finds that the instance reads snapshots for a long time and decides to appoint one of the running instances as the new master. The configuration is changed and the new master goes to RW. After some time, the old master finish reading its snapshot and ~goes to RW as well~ goes to RO, re-reads config, keep itself as RO.

The relevant startup code is [here](https://github.com/tarantool/tarantool/blob/2c422a0b98675cab480ebb44256dfeb72bd2f820/src/box/lua/config/applier/box_cfg.lua#L563-L680). It also requires some separate effort to describe properly.

Reply:
Sound complicated for configuration reference :) Looks like such info should live in main topics, for example:
- https://www.tarantool.io/en/doc/latest/concepts/replication/repl_architecture/
- https://www.tarantool.io/en/doc/latest/dev_guide/internals/replication/replication_server_startup/

And then we can add a corresponding links. What do you think?

Reply:
Sure. Now, when it is pointed that all the information is about the `failover: off` mode the description looks correct.